src/librbd/api/Mirror.cc

```diff
@@ -1262,6 +1231,77 @@ int Mirror<I>::peer_remove(librados::IoCtx& io_ctx, const std::string &uuid) {
                << cpp_strerror(r) << dendl;
     return r;
   }
+
+  map<string, string> images;
+  r = Image<I>::list_images_v2(io_ctx, &images);
```

Comment:
Shouldn't this just be a loop over all mirrored images? Since we remove the peer first, we have a potential failure point where we cannot re-run this operation if it hits a failure. I wonder if we need a "removing" peer state (or similar) 2 phase commit logic.

Reply:
@dillaman Thinking a little more about it. It seems the current implementation to be ok. Note, we ignore ENOENT both in `remove_peer_config_key` and for `cls_client::mirror_peer_remove` so if `peer_site_remove` failed for some reason, the user may always re-run it.